stdlib/public/core/MutableCollection.swift

```diff
@@ -372,6 +377,17 @@ extension MutableCollection {
       _writeBackMutableSlice(&self, bounds: bounds, slice: newValue)
     }
   }
+
+  /// Exchange the values at indices `i` and `j`.
+  ///
+  /// Has no effect when `i` and `j` are equal.
+  @_inlineable
+  public mutating func swapAt(_ i: Index, _ j: Index) {
+    guard i != j else { return }
```

Comment:
This is an interesting nuance. For reference types, this wouldn't be a drop-in replacement for the old `swap`, would it?